android/src/main/java/com/swmansion/gesturehandler/core/NativeViewGestureHandler.kt

```diff
@@ -44,7 +45,11 @@ class NativeViewGestureHandler : GestureHandler<NativeViewGestureHandler>() {
 
   override fun shouldRecognizeSimultaneously(handler: GestureHandler<*>): Boolean {
     // if the gesture is marked by user as simultaneous with other or the hook return true
-    if (super.shouldRecognizeSimultaneously(handler) || hook.shouldRecognizeSimultaneously(handler)) {
+    hook.shouldRecognizeSimultaneously(handler)?.let {
+      return@shouldRecognizeSimultaneously it
+    }
```

Comment:
I'm not actually sure how this will be interpreted since both methods are called `shouldRecognizeSimultaneously`

Reply:
It does return from `shouldRecognizeSimultaneously` function, not the lambda inside `hook`.